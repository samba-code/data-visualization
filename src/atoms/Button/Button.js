import React from "react";
import styled from "styled-components";

import PropTypes from "prop-types";

const StyledButton = styled.button`
  background: ${(props) =>
    props.selected ? props.theme.primaryLight : props.theme.primary};
  color: white;
  padding: 4px 8px;
  border-radius: 3px;
  border: none;
`;

const Button = ({ children, ...props }) => (
  <StyledButton {...props}>{children}</StyledButton>
);

Button.propTypes = {
  selected: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

Button.defaultProps = {
  selected: false,
};

export default Button;
