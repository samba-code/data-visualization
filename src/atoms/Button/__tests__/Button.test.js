import React from "react";
import { render } from "enzyme";

import Button from "../Button";

describe("Button", () => {
  it("should match snapshot", () => {
    const wrapper = render(<Button>Some text</Button>);
    expect(wrapper).toMatchSnapshot();
  });
  it("should match snapshot in selected state", () => {
    const wrapper = render(<Button selected>Some text</Button>);
    expect(wrapper).toMatchSnapshot();
  });
});
