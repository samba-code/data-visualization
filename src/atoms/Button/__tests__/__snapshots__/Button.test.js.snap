// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  class="Button__StyledButton-he5aif-0 ctYaMH"
>
  Some text
</button>
`;

exports[`Button should match snapshot in selected state 1`] = `
<button
  class="Button__StyledButton-he5aif-0 ctYaMH"
  selected=""
>
  Some text
</button>
`;
